src/scenes/UserGalleryPage/UserGalleryCollection.tsx

```diff
@@ -69,34 +69,22 @@ function UserGalleryCollection({ queryRef, collectionRef, mobileLayout }: Props)
   const galleryId = collection.gallery.dbid;
 
   const { showModal } = useModalActions();
-  const { push, asPath } = useRouter();
-  const navigateToUrl = useNavigateToUrl();
+  const router = useRouter();
+
   const unescapedCollectionName = useMemo(() => unescape(collection.name), [collection.name]);
   const unescapedCollectorsNote = useMemo(
     () => unescape(collection.collectorsNote ?? ''),
     [collection.collectorsNote]
   );
 
-  const username = asPath.split('/')[1];
-  const collectionUrl = `${baseUrl}/${username}/${collection.dbid}`;
-
-  const handleViewCollectionClick = useCallback(
-    (event: React.MouseEvent<HTMLElement>) => {
-      navigateToUrl(`/${username}/${collection.dbid}`, event);
-    },
-    [collection.dbid, navigateToUrl, username]
-  );
+  const username = router.query.username as string;
```

Comment:
This is a slightly more robust way of getting the username from the URL than before (parsing it from the URL).

However, it's a little scary to take either approach outside the context of a Next.js page component, where we implicitly expect this component to only be used inside the context of a user page. I'd suggest in cases like this, we pass things like the username in as a prop.

Reply:
great point!